docs/docs/02-configuration/08-object-factories.md

```diff
@@ -0,0 +1,72 @@
+# Object Factories
+
+By default the generated code will instantiate objects using the default parameterless constructor if available.
+If no parameterless constructor is available, Mapperly will try to map to the constructor arguments (see [constructor mappings](./07-ctor-mappings.md)).
+
+Alternatively custom object factories can be used to construct or resolve target objects.
+To make use of object factories create an object factory method inside the mapper class
+and set the `Riok.Mapperly.Abstractions.ObjectFactoryAttribute` attribute.
+An object factory method needs to be parameterless and needs to return a non-void type.
+The first object factory with a matching return type is used to construct the desired type by Mapperly.
+
+:::info
+If an object factory is used for a certain type,
+Mapperly cannot map to init only properties or constructor parameters.
+:::
+
+```csharp
+[Mapper]
+public static partial class CarMapper
+{
+    // highlight-start
+    [ObjectFactory]
+    private CarDto CreateCarDto()
+        => _diContainer.Resolve<CarDto>();
```

Comment:
I don't think that resolving an object from a DI container in a static mapper is a good example.
Maybe change this to
```
   [ObjectFactory]
    private CarDto CreateCarDto()
        => CarDto.CreateFromCustomMethod();
```

Reply:
Improved.